### PR TITLE
[2022.2] Docs: API changes - Part 2

### PR DESCRIPTION
**Pull request by ellioman**
# Purpose of this PR
Going through the public API Docs and tweaking the scope and/or documentation

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 1]
```diff
@@ -19,46 +19,147 @@ public enum ShadowQuality
...
         _2048 = 2048,
+
+        /// <summary>
+        /// Use this for 4096x4096 shadow resolution.
+        /// </summary>
         _4096 = 4096
     }
 
+    /// <summary>
+    /// This controls the size of the Light Cookie texture.
```
Comment (eh-unity) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:54]: **atlas** texture for additional lights (point, spot)

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 2]
```diff
@@ -19,46 +19,147 @@ public enum ShadowQuality
...
         _2048 = 2048,
+
+        /// <summary>
+        /// Use this for 4096x4096 Light Cookie resolution.
+        /// </summary>
         _4096 = 4096
     }
 
+    /// <summary>
+    /// Options for selecting the format for the Light Cookie texture.
```
Comment (eh-unity) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:85]: Affects whether the additional lightcookies are grayscale only or color. Atlas texture format, so additional lights only. Mainlight uses the cookie texture directly.

Comment (eh-unity) [reply com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:85]: Low precision saves memory and bandwidth. Intended for mobile.

File: com.unity.render-pipelines.universal/Runtime/ShaderUtils.cs [hunk 3]
```diff
@@ -3,22 +3,72 @@
...
 namespace UnityEngine.Rendering.Universal
 {
+    /// <summary>
+    /// Options to get a shader path to URP shaders when calling ShaderUtils.GetShaderGUID();
+    /// <see cref="ShaderUtils"/>.
+    /// </summary>
     public enum ShaderPathID
     {
+        /// <summary>
+        /// Use this for URP Lit shader.
```
Comment (eh-unity) [inline com.unity.render-pipelines.universal/Runtime/ShaderUtils.cs:13]: Is ComplexLit missing from this list? Should it be here?
Anyway ComplexLit is a superset of Lit for advanced effects that would need additional GBuffer storage. I is always forward rendered and it's used as deferred fallback (for those effects).

**Review by eh-unity — COMMENTED** [review]
Lgtm. I added a couple improvement suggestions.

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 4]
```diff
@@ -133,10 +284,21 @@ public enum VolumeFrameworkUpdateMode
     /// </summary>
     public enum UpscalingFilterSelection
     {
+        /// <summary>
+        /// Use this to have URP handle upscaling automatically.
```
Comment (gmitrano-unity) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:288]:
```suggestion
        /// Indicates that URP will select an appropriate upscaling filter automatically.
```

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 5]
```diff
@@ -133,10 +284,21 @@ public enum VolumeFrameworkUpdateMode
...
+        /// </summary>
         [InspectorName("Automatic")]
         Auto,
+
+        /// <summary>
+        /// Use this to use Bilinear Filtering when performing upscaling.
```
Comment (gmitrano-unity) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:294]:
```suggestion
        /// Indicates that Bilinear filtering will be used when performing upscaling.
```

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 6]
```diff
@@ -133,10 +284,21 @@ public enum VolumeFrameworkUpdateMode
...
+        /// </summary>
         [InspectorName("Bilinear")]
         Linear,
+
+        /// <summary>
+        /// Use this to use Nearest-Neighbour filtering when performing upscaling.
```
Comment (gmitrano-unity) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:300]:
```suggestion
        /// Indicates that Nearest-Neighbour filtering will be used when performing upscaling.
```

**Review by gmitrano-unity — CHANGES_REQUESTED** [review]
Suggested some changes for the upscaling filter comments. It's mostly wording stuff, but the automatic filter comment now tries to make it clear that it will select one of the existing filters depending on which one is most appropriate rather than doing something else that's not available to the user.

**Comment by ellioman** [comment]
@gmitrano-unity Thanks for the help! I've committed your suggestions so please do a re-review for me :)

**Review by gmitrano-unity — APPROVED** [review]
Updates look good to me! :)

**Review by oleks-k — APPROVED** [review]
Looks good, thanks, Elvar!

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs (hunk 1 above)
Comment (ellioman) [reply com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs]: Added.

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs (hunk 2 above)
Comment (ellioman) [reply com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs]: Added.

**Pull request merged**